src/plugins/vis_type_table/public/components/table_vis_grid_columns.tsx

```diff
@@ -0,0 +1,148 @@
+              rows,
+            },
+            row: rowIndex,
+            column: rawColumnIndex,
+          },
+        ],
+        negate,
+      },
+    });
+  };
+
+  return cols.map((col, colIndex) => {
+    const cellActions = col.filterable
+      ? [
+          ({ rowIndex, columnId, Component, closePopover }: EuiDataGridColumnCellActionProps) => {
+            const filterValue = rows[rowIndex][columnId];
+            const filterContent = col.formatter?.convert(filterValue);
+
+            const filterForValueText = i18n.translate(
+              'visTypeTable.tableVisFilter.filterForValue',
+              {
+                defaultMessage: 'Filter for value',
+              }
+            );
+            const filterForValueLabel = i18n.translate(
+              'visTypeTable.tableVisFilter.filterForValueLabel',
+              {
+                defaultMessage: 'Filter for value: {filterContent}',
+                values: {
+                  filterContent,
+                },
+              }
+            );
+
+            return (
+              filterValue != null && (
+                <Component
+                  onClick={() => {
+                    filterBucket(rowIndex, colIndex, false);
+                    closePopover();
+                  }}
+                  iconType="plusInCircle"
+                  aria-label={filterForValueLabel}
+                  data-test-subj="filterForValue"
+                >
+                  {filterForValueText}
+                </Component>
+              )
+            );
+          },
+          ({ rowIndex, columnId, Component, closePopover }: EuiDataGridColumnCellActionProps) => {
+            const filterValue = rows[rowIndex][columnId];
+            const filterContent = col.formatter?.convert(filterValue);
+
+            const filterOutValueText = i18n.translate(
+              'visTypeTable.tableVisFilter.filterOutValue',
+              {
+                defaultMessage: 'Filter out value',
+              }
+            );
+            const filterOutValueLabel = i18n.translate(
+              'visTypeTable.tableVisFilter.filterOutValueLabel',
+              {
+                defaultMessage: 'Filter out value: {filterContent}',
+                values: {
+                  filterContent,
+                },
+              }
+            );
+
+            return (
+              filterValue != null && (
+                <Component
+                  onClick={() => {
+                    filterBucket(rowIndex, colIndex, true);
+                    closePopover();
+                  }}
+                  iconType="minusInCircle"
+                  aria-label={filterOutValueLabel}
+                  data-test-subj="filterOutValue"
+                >
+                  {filterOutValueText}
+                </Component>
+              )
+            );
+          },
+        ]
+      : undefined;
+
+    const initialWidth = columnsWidth.find((c) => c.colIndex === colIndex);
```

Comment:
Do you think it'd be worthwhile to maintain a map of id->column pairs so that we don't have to iterate through the entire list of columns every time we need to find a specific column? I'm not super familiar with this component but considering how many times `find` is used for columns I wanted to discuss that.

Reply:
Also, why not add the initial width property to the formatted column object as well?

Reply:
For the first question, yeah, I think @joshuarrrr also suggest the same. I have opened a research issue [here](https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2549). The reason is that column id might change. For example, when we add/remove a column, id will change. Therefore, when we format the column, we might fail to find the column if use id. But I agree it is pain. After done the functional and unit test, I will re-visit this enhancement issue.

Reply:
For the second question, not quite sure if my understand of the question is correct. We are changing the formatted col to data grid formatted col here, since EuiDataGrid component requests EuiDataGridColumn format.  EuiDataGridColumn need to have the width info to display col in correct width. In `EuiDataGridColumn`, `InitialWidth` is to store the col width and formatted col doesn't need to store the info.